src/noop.jl

```diff
@@ -97,16 +99,34 @@ function Base.seekend(stream::NoopStream)
     return stream
 end
 
-function Base.unsafe_write(stream::NoopStream, input::Ptr{UInt8}, nbytes::UInt)
+function Base.write(stream::NoopStream, b::UInt8)::Int
+    changemode!(stream, :write)
+    if has_sharedbuf(stream)
+        # directly write data to the underlying stream
+        n = Int(write(stream.stream, b))
```

Comment:
Why do you need the explicit cast to `Int` here?

Reply:
I need this to fix the `stats` function for `NoopStream`, but that's for a future PR.

Reply:
Right now the cast isn't doing anything.